js/bitget.js

```diff
@@ -1751,17 +1751,27 @@ module.exports = class bitget extends Exchange {
         await this.loadMarkets ();
         const market = this.market (symbol);
         const [ marketType, query ] = this.handleMarketTypeAndParams ('cancelOrder', market, params);
-        const method = this.getSupportedMapping (marketType, {
+        let method = this.getSupportedMapping (marketType, {
             'spot': 'privateSpotPostTradeCancelOrder',
             'swap': 'privateMixPostOrderCancelOrder',
         });
         const request = {
             'symbol': market['id'],
             'orderId': id,
         };
+        const stop = this.safeValue (params, 'stop');
+        const planType = this.safeString (params, 'planType');
+        if (stop) {
+            method = 'privateMixPostPlanCancelPlan';
+            if (planType === undefined) {
+                throw new ArgumentsRequired (this.id + ' cancelOrder() requires a planType parameter for stop orders, either normal_plan, profit_plan or loss_plan');
+            }
+            request['planType'] = planType;
+        }
         if (marketType === 'swap') {
             request['marginCoin'] = market['settleId'];
         }
+        params = this.omit (params, [ 'stop', 'planType' ]);
```

Comment:
We can move this line under the if (stop) conditional to avoid executing it on all calls.